launch/launch/actions/execute_process.py

```diff
@@ -716,6 +750,27 @@ async def __execute_process(self, context: LaunchContext) -> None:
             self.__logger.error("process has died [pid {}, exit code {}, cmd '{}'].".format(
                 pid, returncode, ' '.join(cmd)
             ))
+            # respawn the process that abnormally died
+            if not self.__shutdown_received and not context.is_shutdown and self.__respawn:
+                respawn_flag = True
+                if self.__respawn_delay > 0.0:
+                    # wait for a timeout(`self.__respawn_delay`) to respawn the process
+                    # and handle shutdown event with future(`self.__shutdown_future`)
+                    # to make sure `ros2 launch` exit in time
+                    await asyncio.wait(
+                        [asyncio.sleep(self.__respawn_delay), self.__shutdown_future],
+                        loop=context.asyncio_loop,
+                        return_when=asyncio.FIRST_COMPLETED
+                    )
+                    if self.__shutdown_future.done():
+                        respawn_flag = False
+                if respawn_flag:
```

Comment:
@iuhilnehc-ynos hmm, do we need `respawn_flag`? why not

```python
if not self.__shutdown_received and not context.is_shutdown and self.__respawn:
    if self.__respawn_delay > 0.0:
        # ...
    if not self.__shutdown_future.done():
        context.asyncio_loop.create_task(self.__execute_process(context))
        return
```

which somewhat emphasizes the redundancy between `shutdown_received` and `shutdown_future`.

Reply:
@hidmic 
Yes, it seems it can work without an extra flag. And I notice that I can also move the 'create_task' behind.